### PR TITLE
Add the ability to specify loss functions as delegates

### LINKED ISSUE #17: Loss function class
In Pytorch loss functions in torch.nn work as follows:

```
loss = NLLLoss(SOME PARAMETERS IF NEEDED)
prediction = model.forward(input)
output = loss(prediction, target)
```

We should match the behavior where once initialized the loss can be applied to the prediction and the target tensors.

### PR DESCRIPTION
**Pull request by interesaaat**
This PR allows to match the python interface when specifying the loss function.

Fix #17.

### CODE REVIEW

File: TorchSharp/NN/LossFunction.cs [hunk 1]
```diff
@@ -9,14 +9,21 @@ namespace TorchSharp.NN
     /// </summary>
     public class LossFunction
     {
+        public delegate TorchTensor Loss(TorchTensor source, TorchTensor target);
+
         [DllImport("libTorchSharp")]
         extern static IntPtr THSNN_lossBCE(IntPtr srct, IntPtr trgt, IntPtr wgt, long reduction);
 
-        public static TorchTensor BCE<T, U>(TorchTensor src, TorchTensor target, TorchTensor? weigths = null, Reduction reduction = Reduction.Mean)
+        public static TorchTensor BCE(TorchTensor src, TorchTensor target, TorchTensor? weigths = null, Reduction reduction = Reduction.Mean)
```
Comment (artidoro) [inline TorchSharp/NN/LossFunction.cs]: I am not sure if we should keep these?
We woud have two ways of doing the same thing. Maybe we could get rid of them?

Comment (interesaaat) [reply TorchSharp/NN/LossFunction.cs]: Agreed. Let me fix this.

**Review by artidoro — APPROVED** [review]
